clic.adoc

```diff
@@ -377,94 +377,144 @@ The existing timer (`mtip`/`stip`), software
 (`msip`/`ssip`), and external interrupt inputs
 (`meip`/`seip`) are treated as additional local interrupt
 sources, where the privilege mode, interrupt level, and priority can
-be altered using memory-mapped `clicintattr[__i__]` and
+be altered using `clicintattr[__i__]` and
 `clicintctl[__i__]` registers.
 
 NOTE: In CLIC mode, interrupt delegation for these signals is achieved
 via changing the interrupt's privilege mode in the CLIC Interrupt
 Attribute Register (`clicintattr`), as with any other CLIC
 interrupt input.
 
-== smclic M-mode CLIC extension
+== CLIC register access via indirect CSR Access
 
-=== CLIC Memory-Mapped Registers
+Access to CLIC registers clicinttrig[i], clicintip[i], clicintie[i], clicintattr[i], clicintctl[i] 
+is specified using the Indirect CSR Access extension method (Smcsrind/Sscsrind).  A CSR accessible via one
+method may or may not be accessible via the other method such as memory-mapped access.
 
-==== CLIC Memory Map
+The Indirect CSR Access allocated range for CLIC is TBD.  These address provided below are relative to the base of that allocated range.
 
-Each hart has a separate CLIC accessed by a separate address region.
-The M-mode CLIC memory map region must be made accessible to
-the M-mode software running on the hart.
+[%autofit]
 
-NOTE: A bus memory map or locked PMP entries could prevent M-mode
-software on a particular hart from reaching the CLIC memory map.
+----
+ miselect  |     mireg     |  mireg                      |     description    |
+ offset    |     bits      |  register state             |                    |
 
-NOTE: For reserved memory regions, specific trap behavior is not specified. Depending on system bus architecture, the system can ignore the access (e.g., read zero/write ignored) or cause a bus error (usually imprecise interrupt), or some other platform-specific behavior. The "reserved" annotation here implies that future standards might place additional standard registers in that space, and so using the space for non-standard features is inadvisable.
+  0x0      |     31:0     |   R or RW   clicintip[31:0]  |  clic interrupt pending for interrupts 31-0 |
```

Comment:
Should just mark as RW, as we can ignore writes if the interrupt is not edge-triggered.  This way we don't have to try to deal with exceptions on writes to read-only bits in a register.